### PR TITLE
kmsxx: 2020-08-04 -> 2020-12-18

### PR DESCRIPTION
**Pull request by rmcgibbo**
###### Motivation for this change
This build was [broken on hydra](https://nix-cache.s3.amazonaws.com/log/3yc28q0m896xbs4mvv5g3dc7v99hinnr-python3.7-kmsxx-2020-08-04.drv), probably because of a gcc upgrade (?).

I updated to the latest version to pull in fixes for gcc, which switched the build system from cmake -> meson, so that's the cause of most of the churn.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

------------------------

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>kmsxx</li>
    <li>python37Packages.kmsxx</li>
    <li>python38Packages.kmsxx</li>
    <li>python39Packages.kmsxx</li>
  </ul>
</details>

### CODE REVIEW

File: pkgs/development/libraries/kmsxx/default.nix [hunk 1]
```diff
@@ -1,22 +1,36 @@
...
+    sha256 = "0wni7iymf8mj6vymr092l9zq4b73lbm99ad45i5ihcl9y3y4rk2x";
   };
 
-  cmakeFlags = lib.optional (!withPython) "-DKMSXX_ENABLE_PYTHON=OFF";
+  mesonFlags = lib.optional (!withPython) "-Dpykms=disabled";
 
-  nativeBuildInputs = [ cmake pkg-config ];
-  buildInputs = [ libdrm python ];
+  # meson uses CMake to find pybind11, it can't seem to find pybind11 otherwise
+  nativeBuildInputs = [ meson ninja pkg-config pybind11 git cmake ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/kmsxx/default.nix:32]: Did you try if removing pkg-config works?

Comment (rmcgibbo) [reply pkgs/development/libraries/kmsxx/default.nix:32]: Yes, it doesn't build without pkgs-config.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111976` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>kmsxx</li>
<li>python37Packages.kmsxx</li>
<li>python38Packages.kmsxx</li>
<li>python39Packages.kmsxx</li>
</ul>
</details>

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -3504,9 +3504,9 @@ in {
 
   kmapper = callPackage ../development/python-modules/kmapper { };
 
-  kmsxx = toPythonModule ((callPackage ../development/libraries/kmsxx {
-    inherit (pkgs.kmsxx) stdenv;
-    inherit (pkgs) pkg-config;
+  kmsxx = toPythonModule ((pkgs.callPackage ../development/libraries/kmsxx {
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:3507]: I am not sure if this mixes python versions. @jonringer

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111976` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>kmsxx</li>
<li>python37Packages.kmsxx</li>
<li>python38Packages.kmsxx</li>
<li>python39Packages.kmsxx</li>
</ul>
</details>

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (rmcgibbo) [reply pkgs/top-level/python-packages.nix:3507]: I checked with `ldd` and in a python shell -- it's importable and doesn't look like there's evidence in ldd of multiple pythons or anything.

**Comment by FliegendeWurst** [comment]
PR #132950 included this update.

**Pull request closed without merging**